### DISCUSSION TITLE
[Feature Request] Call other saved SQL queries from Dashboard query

### DISCUSSION THREAD

**Discussion by prescience-data** [supabase/supabase#3164] [Feature Requests]
# Feature request

Not exactly sure how this would work from a practical perspective, but it would be awesome if you could `chain` the queries somehow.

## Describe the solution you'd like

Perhaps some settings item attached to each query's context menu that allows users to set `then` triggers. 

For example:

1. Hit `Run` on `Query 1`
2. If successful, check the query's attached `then` items.
3. Run `Query 2` (and in turn if `Query 2` has `Query 3` attached as a `then` item run that, etc).

## Describe alternatives you've considered

Creating a huge master SQL script - however, this is starting to get unwieldy and hard to navigate in my project.

**Comment by darora** [comment]
@prescience-data have you considered using something like DBT ( https://docs.getdbt.com/tutorial/setting-up )?

**Comment by kiwicopple** [comment]
Also check out [Postgres CTEs](https://www.postgresql.org/docs/current/queries-with.html) (which are shown in Div's link).

```sql
with 
query_1 as (
    -- some query which returns a record set
), 
query_1 as (
    -- some query which returns a record set
)
select *
from query_1 join query_2 on query_1.xxx = query_2.xxx
```

(I'll convert this one to a Discussion :) )
